internal/fs/read_cache_test.go

```diff
@@ -0,0 +1,363 @@
+// Copyright 2023 Google Inc. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// A collection of tests for a file system where the file cache is enabled
+// without download-file-for-random-read set to True.
+
+package fs_test
+
+import (
```

Comment:
Should we also add the case, what happens if (a) read sequential via cache, (b) write with/without sync and then read back?

Reply:
(a) We can't confirm that the second read is served from cache, so functional tests should check that. (b) similar to (a) we won't know whether the read is served from cache or not. at fs level.

However, I still added the test case to confirm that read is happening and correct data is served after served write and sync when read cache is enabled.